### DISCUSSION TITLE
Item Textures Show Up As Black

### DISCUSSION THREAD

**Discussion by Quasar500** [bridge-core/bridge.#352] [Add-On Q&A] [answered]
My item texture just appears as a black square. 
@Chikorita-Lover you said you'd help
![2021-05-27 (1)](https://user-images.githubusercontent.com/80136740/119929712-deabc080-bf32-11eb-84f4-0b3179d0328d.png)

**Comment by Chikorita-Lover** [comment, marked as answer]
Hmm, could you send the `item_texture.json` _file_?

**Reply by Quasar500** [reply]
![2021-05-27 (2)](https://user-images.githubusercontent.com/80136740/119930133-b96b8200-bf33-11eb-99b4-2e09976e66ec.png)

**Reply by Quasar500** [reply]
that one?

**Reply by Chikorita-Lover** [reply]
Yes; first, replace **all** instances of `Common Loot Bag` with `common_loot_bag` (that includes **but is not limited to** renaming the image and item behavior files).
Next, add `frame` to the icon component:
```jsonc
{
  "minecraft:icon": {
    "texture": "common_loot_bag", //Remember, this was changed as mentioned above
    "frame": 0
  }
}
```
General rule of thumb for add-ons: **never** use capital letters, except for within the LANG files.

**Reply by Quasar500** [reply]
still didn't work

**Reply by Chikorita-Lover** [reply]
Could you send the item behavior and `item_texture.json` _files_?

**Reply by Quasar500** [reply]
![2021-05-27 (3)](https://user-images.githubusercontent.com/80136740/119930806-06038d00-bf35-11eb-8174-3211e53e1a5e.png)
![2021-05-27 (4)](https://user-images.githubusercontent.com/80136740/119930808-069c2380-bf35-11eb-8879-2ae96d4d4054.png)

**Reply by Chikorita-Lover** [reply]
You haven't updated the `Common Loot Bag` object in `texture_data` (within the `item_texture.json` file).

**Reply by Quasar500** [reply]
it is just see through now, I renamed the Item Common Loot Bag in the texts so updating it broke it

**Reply by Chikorita-Lover** [reply]
Yeah, just use `item.quasar:common_loot_bag=Common Loot Bag` in the LANG files.
As for the texture issue, try adding `category` to `description` and use bridge.'s auto-completions to set a value (it's very self-explanatory when using auto-completions).

**Reply by Quasar500** [reply]
set a value? which value?

**Reply by Chikorita-Lover** [reply]
You're setting a value for `category`; as I said, relying on bridge.'s auto-completions to set the value will make it very understandable/self-explanatory.

**Reply by Quasar500** [reply]
i set it to items, textures still broken, is it something with the image?

**Reply by Chikorita-Lover** [reply]
Hmm, what image type is it? PNG, TGA, JPEG...?

**Reply by Quasar500** [reply]
PNG

**Reply by Chikorita-Lover** [reply]
Mmkay... could you send the item behavior and `item_texture.json` files?

**Reply by Quasar500** [reply]
![2021-05-27 (5)](https://user-images.githubusercontent.com/80136740/119932025-61cf1580-bf37-11eb-9eff-8f73e1f14259.png)
![2021-05-27 (6)](https://user-images.githubusercontent.com/80136740/119932026-6267ac00-bf37-11eb-8b2f-f1c6f4dc00dc.png)

**Reply by Chikorita-Lover** [reply]
> You haven't updated the `Common Loot Bag` object in `texture_data` (within the `item_texture.json` file).

This still applies to what you've just sent

**Reply by Quasar500** [reply]
well I updated it at one point and it made the texture nothing because in the text I've renamed it Common Loot Bag so if i change it it doesn't know what I'm talking about

**Reply by Chikorita-Lover** [reply]
LANG files and `item_texture.json` have nothing to do with each other; LANG files rely solely on the item's identifier and format version.

**Reply by Quasar500** [reply]
If I change it it breaks the code I tried it

**Reply by Chikorita-Lover** [reply]
The value of `texture` (in the icon component) and the object in `item_texture.json` have to match; currently, in what you've sent aboe, they don't.

**Reply by Quasar500** [reply]
look I changed it and this happens
![2021-05-27 (7)](https://user-images.githubusercontent.com/80136740/119933272-a3ac8b80-bf38-11eb-9d82-6b56e04b3e40.png)
![2021-05-27 (9)](https://user-images.githubusercontent.com/80136740/119933275-a4452200-bf38-11eb-8ff7-97c44d557ba5.png)

**Reply by Quasar500** [reply]
the item im holding is the loot bag

**Reply by Chikorita-Lover** [reply]
Mmkay, either way, keep it how it is now.

**Reply by Quasar500** [reply]
but its broken how it is now...

**Reply by Chikorita-Lover** [reply]
The name of the object cannot be the only source of the issue. Have you saved both the item behavior and `item_texture.json` files?

**Reply by Quasar500** [reply]
yes, there has to be a problem getting the texture or something, should i try changing the image type?

**Reply by Chikorita-Lover** [reply]
Stick with the PNG format for now. If you remove the entity placer component, does it work?

**Reply by Quasar500** [reply]
no it doesn't

**Reply by Chikorita-Lover** [reply]
Mmkay, feel free to re-add it. You have the resource pack added to the world, yes?

**Reply by Quasar500** [reply]
what image types or the normal textures in the basic resource pack?

**Reply by Quasar500** [reply]
yes i do

**Reply by Quasar500** [reply]
its added

**Reply by Chikorita-Lover** [reply]
The vanilla resource pack uses PNG images.

**Reply by Chikorita-Lover** [reply]
Try renaming the item behavior file from `Common Loot Bag.json` to `common_loot_bag.json`.

**Reply by Quasar500** [reply]
nope didn't work

**Reply by Chikorita-Lover** [reply]
Hmm, could you send the behavior and resource packs so I can take a better look at what's going on?

**Reply by Quasar500** [reply]
wait i found out something useful

**Reply by Quasar500** [reply]
i don't think it knows what image I'm talking about because when I changed the texture name to a different name then the actual file, it was still a black square, that must mean that the black square means it can't find the file right?

**Reply by Chikorita-Lover** [reply]
I'm not sure what you're talking about. The file path you set in `item_texture.json` is the exact file path of the texture image, right?

**Reply by Quasar500** [reply]
i believe so yea

**Reply by Chikorita-Lover** [reply]
Hmm, could you double check?

**Reply by Quasar500** [reply]
ok so the path was textures/items/common_loot_bag but the items folder was spelt Items with a capital so i fixed it and now its no texture at all instead of a black square

**Reply by Chikorita-Lover** [reply]
Could you send the item behavior and `item_texture.json` files?

**Reply by Quasar500** [reply]
Is bridge glitched?

**Reply by Chikorita-Lover** [reply]
No, you've just done something wrong and we're trying to get to the bottom of this.

**Reply by Quasar500** [reply]
![2021-05-27 (10)](https://user-images.githubusercontent.com/80136740/119934935-7a412f00-bf3b-11eb-8094-f07f09e90ca8.png)
![2021-05-27 (11)](https://user-images.githubusercontent.com/80136740/119934944-7dd4b600-bf3b-11eb-8b0c-24c6c1aaccfe.png)

**Reply by Chikorita-Lover** [reply]
Oh, you need `frame` in the icon component, like this:
```json
{
  "minecraft:icon": {
    "texture": "common_loot_bag",
    "frame": 0
  }
}
```

**Reply by Quasar500** [reply]
theres how it looks
![2021-05-27 (12)](https://user-images.githubusercontent.com/80136740/119935152-defc8980-bf3b-11eb-8bfb-9f0d90907e3b.png)

**Reply by Quasar500** [reply]
did frame still looks like that

**Reply by Chikorita-Lover** [reply]
Could you send the item behavior and `item_texture.json` _files_?

**Reply by Quasar500** [reply]
![2021-05-27 (13)](https://user-images.githubusercontent.com/80136740/119935448-5d592b80-bf3c-11eb-8149-13ec0e814e3a.png)
![2021-05-27 (14)](https://user-images.githubusercontent.com/80136740/119935453-5df1c200-bf3c-11eb-94b6-5bae0271f11e.png)

**Reply by Quasar500** [reply]
i can send you the behavior and resource pack files if needed just pls dont steal

**Reply by Chikorita-Lover** [reply]
You need to rename the `Items` folder (within the `textures` folder) to `items`.

**Reply by Quasar500** [reply]
I already did

**Reply by Chikorita-Lover** [reply]
Then why does the folder appear as `Items` in the screenshots you sent?

**Reply by Quasar500** [reply]
it doesn't, im confused

**Reply by Chikorita-Lover** [reply]
In the resource pack sidebar shown in your above screenshots, the folder within the `textures` folder is `Items`, not `items`.

**Reply by Quasar500** [reply]
that I don't know because look
![2021-05-27 (15)](https://user-images.githubusercontent.com/80136740/119935950-394a1a00-bf3d-11eb-8e57-a00059b538b8.png)

**Reply by Chikorita-Lover** [reply]
Ah, you just hadn't refreshed bridge.

**Reply by Quasar500** [reply]
refreshed

**Reply by Chikorita-Lover** [reply]
Hmm, could you send the behavior and resource packs so I could take a better look?

**Reply by Quasar500** [reply]
nothing appears in the packs but my loot bag picture, besides that its just folders

**Reply by Chikorita-Lover** [reply]
What about the JSON files?

**Reply by Quasar500** [reply]
you cant send .json files in here

**Reply by Chikorita-Lover** [reply]
You can't send JSON files on GitHub? Are you able to send ZIP files for the behavior and resource packs?

**Reply by Quasar500** [reply]
do you have discord or something we can talk on?

**Reply by Chikorita-Lover** [reply]
Yep, I'm in the bridge. Discord server: https://discord.gg/qqUSDvQU

**Reply by Chikorita-Lover** [reply]
> You haven't updated the `Common Loot Bag object` in `texture_data` (within the `item_texture.json` file).

Based on the pack you sent me over Discord, this was the only issue.

**Reply by Chikorita-Lover** [reply]
Here are the updated behavior and resource packs as ZIP archives, in case you had changed anything else since:

[New Craft Behavior Pack 0.1.0.zip](https://github.com/bridge-core/bridge./files/6566653/New.Craft.Behavior.Pack.0.1.0.zip)
[New Craft Resource Pack 0.1.0.zip](https://github.com/bridge-core/bridge./files/6566654/New.Craft.Resource.Pack.0.1.0.zip)
